File: CSharp/ImageProcessing/Lib/ComHistgramLiveCharts.cs
using LiveCharts;
using LiveCharts.Wpf;
using OxyPlot;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Media.Imaging;

/// <summary>
/// ライブチャートのヒストグラム表示のロジック
/// </summary>
class ComHistgramLiveCharts : ComCharts
{
    /// <summary>
    /// ヒストグラム用の2次元配列データ　配列の1次元：オリジナルのデータ、配列の2次元：画像処理後のデータ
    /// </summary>
    public int[,] Histgram
    {
        get { return base.m_nHistgram; }
    }

    /// <summary>
    /// ビットマップ
    /// </summary>
    public BitmapImage Bitmap
    {
        set { base.m_bitmap = value; }
        get { return base.m_bitmap; }
    }

    /// <summary>
    /// Writeableなビットマップ
    /// </summary>
    public WriteableBitmap WBitmap
    {
        set { base.m_wbitmap = value; }
        get { return base.m_wbitmap; }
    }

    /// <summary>
    /// コンストラクタ
    /// </summary>
    public ComHistgramLiveCharts()
    {
    }

    /// <summary>
    /// デスクトラクタ
    /// </summary>
    ~ComHistgramLiveCharts()
    {
    }

    /// <summary>
    /// ヒストグラムの描画データ処理
    /// </summary>
    /// <returns>ヒストグラムのデータ</returns>
    public GraphData DrawHistgram()
    {
        GraphData graphData = new GraphData();

        base.InitHistgram();

        base.CalHistgram();

        var chartValue1 = new ChartValues<int>();
        var chartValue2 = new ChartValues<int>();
        for (int nIdx = 0; nIdx < (base.m_nHistgram.Length >> 1); nIdx++)
        {
            chartValue1.Add(base.m_nHistgram[(int)ComInfo.PictureType.Original, nIdx]);
            if (m_wbitmap == null)
            {
                chartValue2.Add(0);
            }
            else
            {
                chartValue2.Add(base.m_nHistgram[(int)ComInfo.PictureType.After, nIdx]);
            }
        }

        var seriesCollection = new SeriesCollection();

        var lineSeriesChart1 = new LineSeries()
        {
            Values = chartValue1,
            Title = "Original Image"
        };
        seriesCollection.Add(lineSeriesChart1);

        var lineSeriesChart2 = new LineSeries()
        {
            Values = chartValue2,
            Title = "After Image"
        };
        seriesCollection.Add(lineSeriesChart2);

        graphData.seriesCollection = seriesCollection;

        return graphData;
    }
}

File: Vb/ImageProcessing/Lib/ComHistgramLiveCharts.vb
Imports LiveCharts
Imports LiveCharts.Wpf
Imports OxyPlot

''' <summary>
''' ライブチャートのヒストグラム表示のロジック
''' </summary>
Public Class ComHistgramLiveCharts : Inherits ComCharts
    ''' <summary>
    ''' ヒストグラム用の2次元配列データ　配列の1次元：オリジナルのデータ、配列の2次元：画像処理後のデータ
    ''' </summary>
    Public ReadOnly Property Histgram() As Integer(,)
        Get
            Return MyBase.m_nHistgram
        End Get
    End Property

    ''' <summary>
    ''' ビットマップ
    ''' </summary>
    Public Property Bitmap() As BitmapImage
        Set(value As BitmapImage)
            MyBase.m_bitmap = value
        End Set
        Get
            Return MyBase.m_bitmap
        End Get
    End Property

    ''' <summary>
    ''' Writeableなビットマップ
    ''' </summary>
    Public Property WBitmap() As WriteableBitmap
        Set(value As WriteableBitmap)
            MyBase.m_wbitmap = value
        End Set
        Get
            Return MyBase.m_wbitmap
        End Get
    End Property

    ''' <summary>
    ''' コンストラクタ
    ''' </summary>
    Public Sub New()
    End Sub

    ''' <summary>
    ''' デスクトラクタ
    ''' </summary>
    Protected Overrides Sub Finalize()
        MyBase.Finalize()
    End Sub

    ''' <summary>
    ''' ヒストグラムの描画データ処理
    ''' </summary>
    ''' <returns>ヒストグラムのデータ</returns>
    Public Function DrawHistgram() As GraphData
        Dim graphData = New GraphData()

        MyBase.InitHistgram()

        MyBase.CalHistgram()

        Dim chartValue1 = New ChartValues(Of Integer)
        Dim chartValue2 = New ChartValues(Of Integer)

        For nIdx As Integer = 0 To (MyBase.m_nHistgram.Length >> 1) - 1 Step 1
            chartValue1.Add(MyBase.m_nHistgram(ComInfo.PictureType.Original, nIdx))
            If (m_wbitmap Is Nothing) Then
                chartValue2.Add(0)
            Else
                chartValue2.Add(MyBase.m_nHistgram(ComInfo.PictureType.After, nIdx))
            End If
        Next

        Dim seriesCollection = New SeriesCollection()

        Dim lineSeriesChart1 = New LineSeries() With
        {
            .Values = chartValue1,
            .Title = "Original Image"
        }
        seriesCollection.Add(lineSeriesChart1)

        Dim lineSeriesChart2 = New LineSeries() With
        {
            .Values = chartValue2,
            .Title = "After Image"
        }
        seriesCollection.Add(lineSeriesChart2)

        graphData.seriesCollection = seriesCollection

        Return graphData

    End Function
End Class